server/src/PdfGenerator.tsx

```diff
@@ -0,0 +1,29 @@
+import * as React from 'react';
+import * as ReactDOMServer from 'react-dom/server';
+import { ItemGroup } from './Models';
+import { PdfComponent } from './components/PdfComponent';
+import { Stream } from "stream";
+const wkhtmltopdf = require('wkhtmltopdf');
+
+export class HtmlRenderer {
+    static renderBody(items: ItemGroup[], subject: string, grade: string) {
+        return ReactDOMServer.renderToString(
+            <PdfComponent items={items} subject={subject} grade={grade}/>
+        );
+    }
+}
+
+export class PdfGenerator {
+    static generate(html: string, title: string) {
+        const urlTitle = encodeURIComponent(title);
+        const port = process.env.PORT || 3000;
+        const options = {
+            headerHtml: 'http://localhost:' + port + '/pdf-header.html?title=' + urlTitle,
```

Comment:
make sure line 21 wont be different on the production server

Reply:
It should always point to the current server instance. Now we store the port in the config and I forgot to update it, so thanks for pointing it out ;)